src/components/SideBarComponents/SideBar.tsx

```diff
@@ -0,0 +1,67 @@
+"use client";
+import { useState } from "react";
+import styles from "./SideBar.module.css";
+import SideBarItem from "./SideBarItem";
+import SideSearch from "./SideSearch";
+
+function SideBar() {
+  const data = [
+    {
+      category: "Sorting Algorithms",
+      prefixPath: "sorting-algorithms",
+      algorithms: [
+        {
+          name: "Bubble Sort",
+          path: "bubble-sort",
+        },
+        {
+          name: "Merge Sort",
+          path: "merge-sort",
+        },
+        {
+          name: "Stalin Sort",
+          path: "stalin-sort",
+        },
+      ],
+    },
+
+    {
+      category: "Searching Algorithms",
+      prefixPath: "searching-algorithms",
+      algorithms: [
+        {
+          name: "Linear Search",
+          path: "linear-search",
+        },
+      ],
+    },
+  ];
+
+  const [searchData, setSearchData] = useState<string>("");
+  const pullInputData = (inputData: string) => {
```

Comment:
rename functions so they have naming acording to their functionality for example: handleInputChange

Reply:
Done